deepquest/predict.py

```diff
@@ -210,25 +210,47 @@ def check_params(params):
         warnings.warn('It seems that the pretrained word vectors provided for the target text are not in npy format.'
                       'You should preprocess the word embeddings with the "utils/preprocess_*_word_vectors.py script.')
 
+def findDataset(model):
+    import fnmatch
+    datasets=[]
+    for file in os.listdir(os.path.split(model)[0]):
+        if fnmatch.fnmatch(file,'Dataset*.pkl'):
+                datasets.append(file)
+    assert len(datasets)==1, "Expected to find exactly one dataset in {}/ but found {}\n{}\nPlease specify a dataset file.".format(model, len(datasets), datasets)
```

Comment:
`assert` statements get removed if you "compile" Python with `-O`. For that reason, for things that you actually want to get checked, they are better as an `if :` with a `raise` statement:

    if len(datasets) != 1:
        raise Exception("Expected to find exactly one dataset in {}/ but found {}\n{}\nPlease specify a dataset file.".format(model, len(datasets), datasets))

My suggested code here uses the builtin `Exception` class which is the superclass of all ordinary exceptions. You are welcome to use another one, maybe `OSError`, or create your own: `class DatasetError(Exception): pass`. But really it's probably not worth worrying about too much.

Reply:
This block has been removed in the new version of `predict()`.
Mark as resolved.